Source/Core/Core/WiiUtils.cpp

```diff
@@ -74,4 +97,392 @@ bool InstallWAD(const std::string& wad_path)
+  std::pair<IOS::ES::TMDReader, std::vector<u8>> DownloadTMD(const std::string& prefix_url,
+                                                             const TitleInfo& title);
+  std::pair<std::vector<u8>, std::vector<u8>> DownloadTicket(const std::string& prefix_url,
+                                                             const TitleInfo& title);
+  std::optional<std::vector<u8>> DownloadContent(const std::string& prefix_url,
+                                                 const TitleInfo& title, u32 cid);
+
+  UpdateCallback m_update_callback;
+  IOS::HLE::Kernel m_ios;
+  Common::HttpRequest m_http{0};
+};
+
+std::string SystemUpdater::GetDeviceRegion()
+{
+  // Try to determine the region from an installed system menu.
+  const auto tmd = m_ios.GetES()->FindInstalledTMD(TITLEID_SYSMENU);
+  if (tmd.IsValid())
+  {
+    const DiscIO::Region region = tmd.GetRegion();
+    static const std::map<DiscIO::Region, std::string> regions = {
+        {DiscIO::Region::NTSC_J, "JPN"},
+        {DiscIO::Region::NTSC_U, "USA"},
+        {DiscIO::Region::PAL, "EUR"},
+        {DiscIO::Region::NTSC_K, "KOR"},
+        {DiscIO::Region::UNKNOWN_REGION, "EUR"}};
+    return regions.at(region);
+  }
+
+  // Try to determine the region from the setting.txt file.
+  // We check that file after the system menu, because Dolphin likes to change the region
+  // automatically without warning the user when launching a disc game from the game list.
+  SettingsHandler gen;
+  const std::string settings_file_path =
+      Common::GetTitleDataPath(TITLEID_SYSMENU, Common::FROM_CONFIGURED_ROOT) + WII_SETTING;
+  if (File::Exists(settings_file_path) && gen.Open(settings_file_path))
+  {
+    const std::string region = gen.GetValue("AREA");
+    if (!region.empty())
+      return region;
+  }
+
+  // Fall back to EUR.
+  return "EUR";
+}
+
+std::string SystemUpdater::GetDeviceId()
+{
+  u32 ios_device_id;
+  m_ios.GetES()->GetDeviceId(&ios_device_id);
+  return StringFromFormat("%" PRIu64, (u64(1) << 32) | ios_device_id);
+}
+
+constexpr const char* GET_SYSTEM_TITLES_REQUEST_PAYLOAD = R"END(\
+<?xml version="1.0" encoding="UTF-8"?>
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
+  xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <soapenv:Body>
+    <GetSystemUpdateRequest xmlns="urn:nus.wsapi.broadon.com">
+      <Version>1.0</Version>
+      <MessageId>0</MessageId>
+      <DeviceId></DeviceId>
+      <RegionId></RegionId>
+    </GetSystemUpdateRequest>
+  </soapenv:Body>
+</soapenv:Envelope>
+)END";
+
+SystemUpdater::Response SystemUpdater::ParseTitlesResponse(const std::vector<u8>& response) const
+{
+  pugi::xml_document doc;
+  pugi::xml_parse_result result = doc.load_buffer(response.data(), response.size());
+  if (!result)
+    return {};
+
+  const pugi::xml_node node = doc.select_node("//*[local-name()='GetSystemUpdateResponse']").node();
```

Comment:
Does pugi support namespaces/a namespace manager so you can just add the prefix up front and use `//nus:GetSystemUpdateResponse` instead?
Pretty much a "doesn't matter" thing, since the `local-name()` test works just fine, but I just wanna know ;)